src/entities/swap/swapV3/index.ts

```diff
@@ -164,17 +238,152 @@ export class SwapV3 implements SwapBase {
+                    swapCall.limitAmount,
+                    swapCall.deadline,
+                    swapCall.wethIsEth,
+                ),
+                value: this.value(swapCall.limitAmount, swapCall.wethIsEth),
+            };
+        }
+        if (!swapCall.pathLimits)
+            throw Error('V3 BatchSwaps need path limits for call construction');
         return {
             to: this.to(),
-            callData: this.callData(
-                swapCall.limitAmount,
+            callData: this.callDataBatchSwap(
+                swapCall.limitAmount.amount,
+                swapCall.pathLimits,
                 swapCall.deadline,
                 swapCall.wethIsEth,
             ),
             value: this.value(swapCall.limitAmount, swapCall.wethIsEth),
         };
     }
 
+    /**
+     * Returns the call data to be sent to the vault contract for a single token swap execution.
+     * @param limit minAmountOut/maxAmountIn depending on SwapKind
+     * @param deadline
+     * @param wethIsEth
+     * @returns
+     */
+    private callDataSingleSwap(
+        limit: TokenAmount,
+        deadline: bigint,
+        wethIsEth: boolean,
+    ): Hex {
+        let callData: Hex;
+
+        if ('exactAmountIn' in this.swaps) {
+            callData = encodeFunctionData({
+                abi: balancerRouterAbi,
+                functionName: 'swapSingleTokenExactIn',
+                args: [
+                    this.swaps.pool,
+                    this.swaps.tokenIn,
+                    this.swaps.tokenOut,
+                    this.swaps.exactAmountIn,
+                    limit.amount, // minAmountOut
+                    deadline,
+                    wethIsEth,
+                    DEFAULT_USERDATA,
+                ],
+            });
+        } else if ('exactAmountOut' in this.swaps) {
+            callData = encodeFunctionData({
+                abi: balancerRouterAbi,
+                functionName: 'swapSingleTokenExactOut',
+                args: [
+                    this.swaps.pool,
+                    this.swaps.tokenIn,
+                    this.swaps.tokenOut,
+                    this.swaps.exactAmountOut,
+                    limit.amount, // maxAmountIn
+                    deadline,
+                    wethIsEth,
+                    DEFAULT_USERDATA,
+                ],
+            });
+        } else throw new Error('Incorrect V3 Swap');
+
+        return callData;
+    }
+
+    /**
+     * Returns the call data to be sent to the vault contract for batchSwap execution.
+     * @param limitAmount total minAmountOut/maxAmountIn depending on SwapKind
+     * @param pathLimits individual path minAmountOut/maxAmountIn depending on SwapKind
+     * @param deadline
+     * @param wethIsEth
+     * @returns
+     */
+    private callDataBatchSwap(
+        limitAmount: bigint,
+        pathLimits: bigint[],
+        deadline: bigint,
+        wethIsEth: boolean,
```

Comment:
How about we set default values for these? 
- Deadline -> MAX
- wethIsEth -> false

I'd expect very few users to go without these default values, so we might be providing a better experience by filling out the value for them.

Reply:
I wasn't really sure about defaults. I was leaning more to making them explicit requirements so the user understands what is actually happening. Not super tied to the idea though so happy to go with what you prefer.

Reply:
✅